github4s/src/main/scala/github4s/algebras/Repositories.scala

```diff
@@ -69,6 +69,20 @@ trait Repositories[F[_]] {
       headers: Map[String, String] = Map()
   ): F[GHResponse[List[Repository]]]
 
+  /**
+   * Search all repositories on GitHub.
+   *
+   * @param query The query string for the search.
+   * @param searchParams List of search params.
+   * @param headers Optional headers to include in the request.
+   * @return A GHResponse with the result of the search
+   */
+  def searchRepos(
+      query: String,
+      searchParams: List[SearchParam],
+      headers: Map[String, String] = Map.empty
```

Comment:
Hi! Thanks a lot for the PR!
Seems like [pagination](https://docs.github.com/en/rest/reference/search#search-repositories) argument is missing for this one... Or it's covered by SearchParams somehow?

Reply:
Ah I just forgot about it. I've added it so it matches other parts of the algebras. Truthfully, I started with a copy of the issues search function which did not have a pagination parameter (maybe it should?) and that's the only reason I missed it.